grakn-engine/src/main/java/ai/grakn/engine/controller/ConceptController.java

```diff
@@ -89,9 +91,47 @@ public ConceptController(EngineGraknTxFactory factory, Service spark,
         spark.get(WebPath.KEYSPACE_RULE, this::getRules);
         spark.get(WebPath.KEYSPACE_ROLE, this::getRoles);
 
+        spark.get(WebPath.CONCEPT_ATTRIBUTES, this::getAttributes);
+        spark.get(WebPath.CONCEPT_KEYS, this::getKeys);
+
         spark.get(WebPath.TYPE_INSTANCES, this::getTypeInstances);
     }
 
+    private String getKeys(Request request, Response response) throws JsonProcessingException {
+        return getAttributes(request, response, true);
+    }
+
+    private String getAttributes(Request request, Response response) throws JsonProcessingException {
+        return getAttributes(request, response, false);
+    }
+
+    private String getAttributes(Request request, Response response, boolean isKey) throws JsonProcessingException {
+        response.type(APPLICATION_JSON);
+
+        Keyspace keyspace = Keyspace.of(mandatoryPathParameter(request, KEYSPACE_PARAM));
+        ConceptId conceptId = ConceptId.of(mandatoryPathParameter(request, ID_PARAMETER));
+
+        try (GraknTx tx = factory.tx(keyspace, READ); Timer.Context context = labelGetTimer.time()) {
+            ai.grakn.concept.Concept concept = tx.getConcept(conceptId);
+
+            //If the concept was not found or is not a thing there are no results to return;
+            if(concept == null || !concept.isThing()){
+                response.status(SC_NOT_FOUND);
+                return "[]";
+            }
+
+            //Get the attributes including keys if needed
+            Stream<Attribute<?>> attributes;
+            if(isKey){
```

Comment:
It might be clearer to pass in the method to get the attributes/keys rather than a boolean.